### PR TITLE
feat(python): add more edge-case indent pattern

### PR DESCRIPTION
**Pull request by lucario387**
- [x] match-case indents: https://docs.python.org/3/whatsnew/3.10.html
- [x] break-continue should dedent
~raise Exception dedent~ moved to discussion in #4557 
~fix return dedent~ moved to discussion in #4557

### CODE REVIEW

File: queries/python/indents.scm
Comment (georgeharker) [inline queries/python/indents.scm:1]: Agreed.

**Review by georgeharker — APPROVED** [review]
This looks good and is an improvement - this should go it, and we should try and figure out the details of whether it's node start or end that things are supposed to attach to separately to this.

File: queries/python/indents.scm
Comment (lucario387) [reply queries/python/indents.scm:1]: I'm kindof trying to do return and raise Exceptions indent too, but yeah at the moment this is fine

**Comment by lucario387** [comment]
@amaanq This resolves the 1st and 2nd request of #4554, the 3rd and 4th is much more than I expected so I will move it to a different PR

**Comment by amaanq** [comment]
Hmm I would've thought Exceptions would be easy, is this fine to merge as is?

**Comment by lucario387** [comment]
> Hmm I would've thought Exceptions would be easy, is this fine to merge as is?

Exceptions have
`raise Exception`
and 
`raise Exception(...)`
I won't push a fix just for the first case. The second one is align indented so needs a fix/change with that in mind

**Comment by lucario387** [comment]
> Hmm I would've thought Exceptions would be easy, is this fine to merge as is?

Well it's a better than before state, and doesn't break anything else
With exceptions I expect it to be a little bit more breaking than usual so I will move it to a different PR to let this be accepted

**Comment by amaanq** [comment]
I don't often see Exceptions with some code/message in it spanning multiple lines, but I could always be wrong.

**Comment by lucario387** [comment]
> I don't often see Exceptions with some code/message in it spanning multiple lines, but I could always be wrong.

I also never use multiline Exceptions, but just to be safe, it'll be out of the scope of this PR

**Review by amaanq — APPROVED** [review]

**Comment by georgeharker** [comment]
> I don't often see Exceptions with some code/message in it spanning multiple lines, but I could always be wrong.

I don't often either, but have done when the exception grabs a whole load of context to take along with it.

**Pull request merged**